deployments/consul-deployment.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: extensions/v1beta1
```

Comment:
are you able to run the edgex services with consul ? I faced errors when I deployed consul . kube-dns was repeatedly exiting. That's why I avoided using consul and instead used kubernetes services for service discovery.Kubernetes does service discovery using kube-dns and kube-proxy , so consul isn't needed .

Reply:
Hi, yes; without core-consul, core-command and other services failed.

Reply:
can you post the output of the errors ? also can you post the status of all services after running consul ?

Reply:
vagrant@controller03:~/edgex-on-kubernetes$ kubectl get pods
NAME                                           READY     STATUS    RESTARTS   AGE
edgex-core-command-555bb7c88b-lqnrj            1/1       Running   0          20m
edgex-core-consul-55884b8d68-wbp9d             1/1       Running   0          21m
edgex-core-data-5f794999bd-2ns5c               1/1       Running   0          20m
edgex-core-metadata-5697764b49-q7ksd           1/1       Running   0          20m
edgex-export-client-676758f7cc-hwndm           1/1       Running   0          20m
edgex-export-distro-589b887c56-dgbs8           1/1       Running   0          20m
edgex-mongo-7b789ddcfc-d44wz                   1/1       Running   0          21m
edgex-support-logging-6b6b9bbc-z4pmj           1/1       Running   0          21m
edgex-support-notifications-5f79467d5b-m92cv   1/1       Running   0          21m
edgex-support-scheduler-565b6d6d57-49hb8       1/1       Running   0          20m
rulesengine-576d5c4bbd-6tz9l                   1/1       Running   0          19m
vagrant@controller03:~/edgex-on-kubernetes$

Reply:
All services running.

Reply:
vagrant@controller03:~/edgex-on-kubernetes$ curl -i -H GET http://10.233.41.32:48081/api/v1/ping
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Content-Type: text/plain;charset=UTF-8
Content-Length: 4
Date: Thu, 21 Jun 2018 23:23:09 GMT

pongvagrant@controller03:~/edgex-on-kubernetes$

vagrant@controller03:~/edgex-on-kubernetes$ curl -i -H GET http://10.233.33.65:48075/api/v1/ping
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Content-Type: text/plain;charset=UTF-8
Content-Length: 4
Date: Thu, 21 Jun 2018 23:26:28 GMT

pongvagrant@controller03:~/edgex-on-kubernetes$

Reply:
Error after fixing container image tags:

vagrant@controller03:~/edgex-on-kubernetes$ kubectl logs edgex-core-command-658b64ff5-pvbv2
...
...
[2018-06-18 20:01:49.691] boot - 6 ERROR [main] --- ConsulPropertySourceLocator: Fail fast is set and there was an error reading configuration from consul.
[2018-06-18 20:01:49.692] boot - 6 ERROR [main] --- SpringApplication: Application startup failed
com.ecwid.consul.transport.TransportException: java.net.UnknownHostException: edgex-core-consul: Name does not resolve
...
...

Reply:
why do we need consul if we can resolve services using kube-proxy and kube-dns  ?

Reply:
I have described about this in my blog : 
https://www.edgexfoundry.org/blog/2018/01/30/huawei-runs-edgex-foundry-kubernetes/